components/UI/LoadingSpinner.js

```diff
@@ -1,10 +1,10 @@
 import styled from 'styled-components';
 
 const LoadingSpinner = styled.div`
-  display: inline-block;
+  display: block;
   width: 2rem;
   height: 2rem;
-  border-color: ${props => props.inverted ? 'var(--color-bg) transparent var(--color-bg) transparent' : 'var(--color-fg) transparent var(--color-fg) transparent '};
+  border-color: ${props => props.inverted ? 'var(--color-background) transparent var(--color-background) transparent' : 'var(--color-foreground) transparent var(--color-foreground) transparent '};
```

Comment:
Generally: This is a good example for a commit that should happen on the `main` branch. The variables have changed there already; it's not really part of this "Refactor navigation" change....